dotnet/src/SemanticKernel.Core/PromptTemplate/KernelPromptTemplate.cs

```diff
@@ -136,5 +139,29 @@ internal IList<Block> RenderVariables(IList<Block> blocks, KernelArguments? argu
             ? block
             : new TextBlock(((ITextRendering)block).Render(arguments), this._loggerFactory)).ToList();
     }
+
+    /// <summary>
+    /// Validates that all the KernelArguments are of string type.
+    /// </summary>
+    /// <param name="arguments">The collection of KernelArguments to validate.</param>
+    /// <exception cref="KernelException">Thrown when an argument is not of type string.</exception>
+    private static void AssertArgumentOfStringType(KernelArguments? arguments)
+    {
+        if (arguments == null)
+        {
+            return;
+        }
+
+        foreach (var argument in arguments)
+        {
+            if (argument.Value is string)
+            {
+                continue;
+            }
```

Comment:
Shouldn't this go through the same TypeConverter logic as with KernelFunctionFromMethod, just with a destination type for now of string?

Reply:
Yes, certainly, it can be done this way. However, I have not done it for two reasons - I would like to keep this PR relatively short, and I plan to add non-string support to the kernel prompt template in one of the following PRs.